File: lua/hamming/hamming.lua
local Hamming = {}

function Hamming.compute(a, b)
    if #a ~= #b then
        return -1
    end
    local count = 0
    for i = 1, #a do
        if a:sub(i, i) ~= b:sub(i, i) then
            count = count + 1
        end
    end
    return count
end

return Hamming

File: go/hamming/hamming.go
// Package hamming provides a function for the calculating the Hamming distance
package hamming

import "errors"

/* Distance returns the Hamming distance between two DNA strands.
If the given strands are of different length it throws and error. */
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("strands must be of the same length")

	}
	count := 0
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			count++
		}
	}
	return count, nil

}
